synapse/lib/link.py

```diff
@@ -154,17 +155,36 @@ def getTlsPeerCn(self):
                 if name == 'commonName':
                     return valu
 
-    def getSpawnInfo(self):
+    async def getSpawnInfo(self):
         info = {}
 
         # selectively add info for pickle...
         if self.info.get('unix'):
             info['unix'] = True
 
+        if self.info.get('tls'):
+            info['unix'] = True
+            link0, sock = await linksock()
+
+            async def relay(link):
+                while True:
+                    byts = await link.recv(1024)
+                    if not byts:
+                        break
+                    await self.send(byts)
+                await link.fini()
+
+                if not self.isfini:
+                    await self.fini()
+
+            asyncio.create_task(relay(link0))
```

Comment:
May want self.schedCoro() here to make sure we get cancelled

Reply:
updated